src/main/java/org/primefaces/component/datatable/export/DataTableExporter.java

```diff
@@ -337,6 +338,66 @@ public void export(FacesContext context, List<DataTable> tables, ExportConfigura
      */
     protected abstract void doExport(FacesContext facesContext, DataTable table, ExportConfiguration config, int index) throws IOException;
 
+    protected abstract String getContentType();
+
+    protected void sendExport2Client(String filenameWithExtension, ByteArrayOutputStream content, FacesContext context) throws IOException {
+        if (PrimeFaces.current().isAjaxRequest()) {
+            ajaxDownload(filenameWithExtension, content.toByteArray(), context);
+        }
+        else {
+            ExternalContext externalContext = context.getExternalContext();
+            externalContext.setResponseContentType(getContentType());
+            setResponseHeader(externalContext, ComponentUtils.createContentDisposition("attachment", filenameWithExtension));
+            externalContext.setResponseContentLength(content.size());
+            addResponseCookie(context);
+
+            content.writeTo(externalContext.getResponseOutputStream());
+            externalContext.responseFlushBuffer();
+        }
+    }
+
+    protected void sendExport2Client(String filenameWithExtension, StringBuilder content, String encodingType, FacesContext context) throws IOException {
+        if (PrimeFaces.current().isAjaxRequest()) {
+            ajaxDownload(filenameWithExtension, content.toString().getBytes(encodingType), encodingType, context);
+        }
+        else {
+            ExternalContext externalContext = context.getExternalContext();
+            externalContext.setResponseContentType(getContentType() + "; charset=" + encodingType);
+            setResponseHeader(externalContext, ComponentUtils.createContentDisposition("attachment", filenameWithExtension));
+
+            addResponseCookie(context);
+
+            OutputStream os = externalContext.getResponseOutputStream();
+            OutputStreamWriter osw = new OutputStreamWriter(os, encodingType);
+            PrintWriter writer = new PrintWriter(osw);
+            writer.write(content.toString());
+            writer.flush();
+            writer.close();
```

Comment:
I don't see your close the outputStreamWriter.  A better way may be try with resources.
```java
        try (OutputStreamWriter osw = new OutputStreamWriter(os, encodingType);
                    PrintWriter writer = new PrintWriter(osw);) {
            writer.write(content.toString());
            writer.flush();
        }
```

Reply:
Good point. Fixed. (Just migrated old and not so perfect code before.)